PluralKit.Bot/Commands/Misc.cs

```diff
@@ -247,5 +252,79 @@ public async Task GetMessage(Context ctx)
 
             await ctx.Reply(embed: await _embeds.CreateMessageInfoEmbed(message));
         }
+
+        public async Task MessageProxyCheck(Context ctx)
+        {
+            ulong channelId;
+            ulong messageId;
+
+            // if the message isn't a reply, check for a message link
+            if (ctx.Message.Type != Message.MessageType.Reply || ctx.Message.MessageReference == null)
+            {
+                if (!ctx.HasNext())
+                    throw new PKSyntaxError("You must reply to a message or pass a message link.");
+
+                var word = ctx.PeekArgument();
+                var match = Regex.Match(word, "https://(?:\\w+.)?discord(?:app)?.com/channels/\\d+/(\\d+)/(\\d+)");
+                if (!match.Success)
+                    throw new PKError($"Could not parse {ctx.PeekArgument().AsCode()} as a message link.");
+                    
+                channelId = ulong.Parse(match.Groups[1].Value);
+                messageId = ulong.Parse(match.Groups[2].Value);
+            } else
+            // message is a reply, use it's values instead
+            {
+                channelId = ctx.Message.MessageReference.ChannelId.Value;
+                messageId = ctx.Message.MessageReference.MessageId.Value;
+            }
+            
+            // get the message info
+            var msg = ctx.Message;
+            try 
+            {
+                msg = await _rest.GetMessage(channelId, messageId);
+            }
+            catch (ForbiddenException)
+            {
+                throw new PKError("Unable to get the message provided.");
+            }
+
+            if (msg == null) 
+                throw new PKError("The message you are checking does not exist.");
+
+            // check if the message is not a webhook and sent by the same user as the message that's being checked
+            if (msg.WebhookId != null)
+                throw new PKError("You cannot check messages sent by a webhook.");
+            if (msg.Author.Id != ctx.Author.Id)
+                throw new PKError("You can only check your own messages.");
+
+
+            // get the channel info
+            var channel = await _rest.GetChannel(msg.ChannelId);
```

Comment:
This should be getting the channel from cache, no need for an API request here (just fail if it's not found)